src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -647,11 +648,10 @@ public void run() {
                         numRetries = 0;
                     }
                 } catch (IOException e) {
-                    if (shutdown) {
-                        break;
-                    }
                     LOG.error("Exception while listening", e);
-                    numRetries++;
+                    if (!(e instanceof SocketTimeoutException)) {
```

Comment:
- update

- l checked the native method `java.net.PlainSocketImpl.socketAccept(Native Method)` in [openjdk](http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/9d617cfd6717/src/solaris/native/java/net/PlainSocketImpl.c), **line709-721**, in which it changed from 0 to -1. and then timeout of -1 is interpreted as an infinite timeout.  In some cases, [-1 was interpreted as a larger positive integer](https://lwn.net/Articles/483078/). so this issue always happend after 49days. It's my wild conjecture.